lib/publish.js

```diff
@@ -1,167 +1,105 @@
 'use strict'
 
-const BB = require('bluebird')
-
-const cacache = require('cacache')
-const figgyPudding = require('figgy-pudding')
-const libpub = require('libnpmpublish').publish
-const libunpub = require('libnpmpublish').unpublish
-const lifecycle = BB.promisify(require('./utils/lifecycle.js'))
+const util = require('util')
 const log = require('npmlog')
-const npa = require('npm-package-arg')
-const npmConfig = require('./config/figgy-config.js')
-const output = require('./utils/output.js')
-const otplease = require('./utils/otplease.js')
-const pack = require('./pack')
-const { tarball, extract } = require('pacote')
-const path = require('path')
-const readFileAsync = BB.promisify(require('graceful-fs').readFile)
-const readJson = BB.promisify(require('read-package-json'))
 const semver = require('semver')
-const statAsync = BB.promisify(require('graceful-fs').stat)
+const pack = require('libnpmpack')
+const libpub = require('libnpmpublish').publish
+const runScript = require('@npmcli/run-script')
+
+const npm = require('./npm')
+const output = require('./utils/output')
+const otplease = require('./utils/otplease')
+const { getContents, logTar } = require('./utils/tar')
```

Comment:
We usually add the `.js` on internal require()s, to cut down on unnecessary stat calls.